Dockerfile

```diff
@@ -108,5 +88,19 @@ HEALTHCHECK --interval=60s --timeout=10s \
     --data-raw '{"operationName":"CheckInitialSetup","variables":{},"query":"query CheckInitialSetup { siteInfo { initialSetup }}"}' \
     || exit 1
 
+# Required dependencies
+COPY scripts/install_runtime_dependencies.sh /app/scripts/
+RUN chmod +x /app/scripts/install_runtime_dependencies.sh \
+  # Create a user to run Photoview server
+  && groupadd -g 999 photoview \
+  && useradd -r -u 999 -g photoview -m photoview \
+  # Required dependencies
+  && /app/scripts/install_runtime_dependencies.sh
+
 USER photoview
 ENTRYPOINT ["/app/photoview"]
+WORKDIR /home/photoview
+
+COPY api/data /app/data
+COPY --from=ui /app/ui/dist /app/ui
+COPY --from=api /app/api/photoview /app/photoview
```

Comment:
Why did you move lines 91-99 and 102-106 down the file? It looks messy now.
Usually, the structure is:
- copy data to the image
- run some commands
- configure the app (including the workdir if not needed to be set earlier for the RUN)
- expose port
- define healthcheck
- declare the user
- declare the entrypoint

Reply:
The previous part won't change frequently. Moving copying binaries down lets Docker cache the previous part between buildings. See [here](https://docs.docker.com/build/cache/).

Reply:
There is no need to cache anything in the Expose, Healthcheck, User, and Entrypoint declarations. You can safely move them down according to the order, I mentioned. There is nothing to cache in Env declarations as well, but it is ok to have them also at the beginning of the stage, as this is the common place for them.
I see the point for reordering Copy and Run calls and it is ok to be ordered in this way, but I'd like to keep other declarative instructions, which don't impact on cache, to be located in the standard order

Reply:
Even those Expose steps don't have too many sizes, they still create new layers for them. From my experience, there is no common structure in a Dockerfile, just ordering with the change frequency. I hope to keep the most important part at the end of the spec.

Reply:
Could you please measure the time difference between the current and the latest proposed by me ordering? I expect to see no difference. If my expectation is true, I'd recommend following the common ordering. (If I remember correctly, I even read some recommendations to keep the User and Entrypoint (or CMD) to be the last 2 declarations)

Reply:
Well, I don't insist on this. Revert as previous steps. The steps below copying won't take much time. I reverted it to the previous order.